### PR TITLE
Add min_score option, with result test!

### PR DESCRIPTION
**Pull request by st0012**
I found the feature in #292 would be very helpful to me, so I add result test and rebase it. Hope this PR can be merge ASAP, thanks 😁.

### CODE REVIEW

**Comment by st0012** [comment]
@ankane Hello ~ is there any problem in my code?

**Comment by ankane** [comment]
Same ask as #292 - can you add tests on how these options affect the search results rather than how they modify the request?

File: test/query_test.rb [hunk 1]
```diff
@@ -11,4 +11,15 @@ def test_basic
     assert_equal ["Apple", "Milk"], query.execute.map(&:name).sort
   end
 
+  def test_with_effective_min_score
+    store_names ["Milk", "Milk2"]
+
+    assert_equal ["Milk"], Product.search("Milk", min_score: 0.1).map(&:name)
```
Comment (dlackty) [inline test/query_test.rb]: @ankane I thought this test and the following one shows "how these options affect the search results"? cc @st0012

Comment (ankane) [inline test/query_test.rb]: These tests do, but the facet and sql tests do not.

Comment (st0012) [inline test/query_test.rb]: @ankane I added facets' result test, but I am not sure what you want to test in sql test. Won't it be a lot like query test?

**Pull request closed without merging**